docs/src/references/c8y-configuration-management.md

```diff
@@ -73,11 +73,20 @@ The `c8y_configuration_plugin` configuration is stored by default under `/etc/te
 This [TOML](https://toml.io/en/) file defines the list of files to be managed from the cloud tenant.
 Each configuration file is defined by a record with:
 * The full `path` to the file.
+
+  By default it is treated as a local filesysten path.
+
+  If it contains the prefix `mqtt://` it is treated as MQTT topic structure, where another process/external device is expected to provide/consume the configuration file.
+  
+  On file downloaded from the cloud to the device, the `c8y_configuration_plugin` publishs the content to the MQTT topic `{path}`, instead of writing it to a local filesystem path.
+  
+  On file upload to the cloud, the `c8y_configuration_plugin` publishs an empty message to the MQTT topic `{path}/retrieval` to request the process/external device to publish the current configuration file content to the MQTT topic `{path}`. Then `c8y_configuration_plugin` consumes the content from MQTT topic `{path}` instead of reading a local filesystem path.
```

Comment:
I understand the need but this would be really fragile. Notably one would have to handle timeouts.

* Did you consider other alternatives?
* This proposal is not aligned with the current design. To keep things both simple and extensible - this configuration manager just notify over MQTT that a new configuration is available. Why not propose something along these lines?
* There is already a [discussion on this topic](https://github.com/thin-edge/thin-edge.io/discussions/1144). It might be good to compare this proposal with those given there.
* It would also be good to have something consistent with [what is discussed for log management](https://github.com/thin-edge/thin-edge.io/pull/1102#discussion_r861176527).

I also have comments on the structure of this doc.
* It would be better to keep this paragraph focused on the content of the configuration file. How these parameters are used should be detailed elsewhere say "configuration files for child devices".
* The issue of managing configuration files for child devices is far more complex than for the main device. Hence, it might be good to add a design document stating the problem and exploring some alternative solutions with their pros and cons.

Reply:
Using retained messages is indeed an option. The main issue is - as you highlighted, that the child devices must give their initial configuration before hands.

Using HTTP is another one. With the same issue regarding initialization. On configuration changed from the cloud, the configuration plugin serves the new configuration over HTTP and notifies the url in the message sent over the `tedge/configuration_change` topic e.g `{ "path": "/etc/tedge/tedge.toml" , "url": "http://192.168.1.6/tedge/configurations/etc/tedge/tedge.toml" }`.

To move forward, it would be good to describe these solutions (at least 3) with the pros and cons. One also needs to make the problem clear e.g "how to deliver files to child devices" and "how to fetch files form child devices".

Reply:
> I also have comments on the structure of this doc.
> 
> * It would be better to keep this paragraph focused on the content of the configuration file. How these parameters are used should be detailed elsewhere say "configuration files for child devices".

I moved child-device details now into the new section 'Configuration files for child devices'. 

I will add there also both other two proposals we discussed above, for providing/consuming config files via network (retain MQTT message, and HTTP instead of MQTT retain messages).

Reply:
> I understand the need but this would be really fragile. Notably one would have to handle timeouts.

I agree. We really should not go into that direction.
As I added now another proposal based in MQTT with retain messages (see comment https://github.com/thin-edge/thin-edge.io/pull/1236#discussion_r922271484 above) I removed that 1st MQTT proposal that uses a request-response pattern.

Reply:
> Using HTTP is another one.  [...]  On configuration changed from the cloud, the configuration plugin serves the new configuration over HTTP [...]

Due to your feedback I added now another proposal based on HTTP. However I am not convinced that this is really what we want. Especially from implementation POV it introduces complexity without much benefit.